### ISSUE TITLE
pygame SDL Import Error when trying to build with cx_freeze on Ubuntu 16.04

### ISSUE THREAD

**Issue by mai1x9** [marcelotduarte/cx_Freeze#1312]
When pygame is build with cx_freeze, and the executable is run, it throws the following error,
```
Traceback (most recent call last):
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/initscripts/__startup__.py", line 40, in run
    module.run()
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/initscripts/Console.py", line 37, in run
    exec(code, {'__name__': '__main__'})
  File "test.py", line 2, in <module>
  File "/home/sai/.local/lib/python3.5/site-packages/pygame/__init__.py", line 81, in <module>
    from pygame.base import * # pylint: disable=wildcard-import; lgtm[py/polluting-import]
ImportError: libSDL2-2-7159ff74.0.so.0.16.0: cannot open shared object file: No such file or directory

```

**More Details:**
----
**OS: Ubuntu 16.04**
**python version: 3.5.2**
**pygame version: 2.0.3**
**cx_freeze version: 6.1**

Python script:
```

import pygame
import pygame.camera
import pygame
import pygame.image

pygame.init()
pygame.camera.init()
cameras = pygame.camera.list_cameras()

print(cameras)
```

**setup.py file**
```
from cx_Freeze import setup, Executable
import os, shutil

packages = ["sqlite3", "pygame"]  #'gobject']


setup(name="Pygame Camera example",
      version="0.1.3",
      description="Camera App",
      options = {
        'build_exe': {
            'packages': packages
        },

    },
    executables=[Executable("test.py")])

```

I have tried to install sdl by running,
```
sudo apt-get install libsdl2-mixer-2.0-0 libsdl2-image-2.0-0 libsdl2-2.0-0
sudo apt-get install python3-sdl2
sudo apt install libsdl2-ttf-2.0-0
```
and tried to re build, still the same issue

Any help would be great

**Comment by marcelotduarte** [comment]
Try with cx_Freeze 6.9
EDIT: For python 3.5.2 you can try with cx_Freeze 6.2 and 6.3

**Comment by mai1x9** [comment]
@marcelotduarte , thanks for quick reply.
**looks like cx freeze 6.3 is not working for ubuntu 16.04**
When i run setup.py as
```
python3 setup.py build
```
strangely it is executing the python the is suppose to be froze. That is it is running test.py file instead of freezing it.(test.py is the Executable to be made)

Below is the output when setup.py is build.
```
running build
running build_exe
pygame 2.0.3 (SDL 2.0.16, Python 3.5.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py:305: VisibleDeprecationWarning: zmq.eventloop.minitornado is deprecated in pyzmq 14.0 and will be removed.
    Install tornado itself to use zmq with the tornado IOLoop.
    
  module = self._InternalImportModule(name, deferredImports)
/usr/local/lib/python3.5/dist-packages/IPython/kernel/__init__.py:13: ShimWarning: The `IPython.kernel` package has been deprecated since IPython 4.0.You should import from ipykernel or jupyter_client instead.
  "You should import from ipykernel or jupyter_client instead.", ShimWarning)
SQLITE imported 
ok
['/dev/video0']
[=>]Resolution Found  (854, 480)
Starting the web camera ..............
webcam closed ###################################
^Cinterrupted

```
and there is no build folder .

The output of test.py when run using 
```
python3 test.py
```
```
pygame 2.0.3 (SDL 2.0.16, Python 3.5.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
# these statements are printed in testpy, and actually testpy got executed without building the file
SQLITE imported 
ok
['/dev/video0']
[=>]Resolution Found  (854, 480)
Starting the web camera ..............
webcam closed ###################################
```
Basically, cx freeze 6.3, it is running test.py as output of it can be seen clearly without building it. I have not changed setup.py file. **The same setup file which I used for cx freeze 6.1, i have used it for cx freeze 6.3.**


----

**When using cx freeze 6.2, it says**
```
sai@sai-cryptaker:~/Desktop/pygame_example$ python3 setup.py build 
running build
running build_exe
Traceback (most recent call last):
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 397, in _InternalImportModule
    spec = importlib.util.find_spec(name)
  File "/usr/lib/python3.5/importlib/util.py", line 102, in find_spec
    raise ValueError('{}.__spec__ is None'.format(name))
ValueError: __main__.__spec__ is None

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup2.py", line 18, in <module>
    executables=[Executable("test.py")])
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/dist.py", line 342, in setup
    distutils.core.setup(**attrs)
  File "/usr/lib/python3.5/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.5/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/lib/python3.5/distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "/usr/lib/python3.5/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/dist.py", line 217, in run
    freezer.Freeze()
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/freezer.py", line 638, in Freeze
    self.finder = self._GetModuleFinder()
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/freezer.py", line 372, in _GetModuleFinder
    self.constantsModule, self.zipIncludes)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 144, in __init__
    self._AddBaseModules()
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 155, in _AddBaseModules
    self.IncludeModule("traceback")
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 653, in IncludeModule
    namespace = namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 309, in _ImportModule
    deferredImports, namespace = namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 410, in _InternalImportModule
    parentModule, namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 473, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 589, in _ScanCode
    module, relativeImportIndex)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 309, in _ImportModule
    deferredImports, namespace = namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 410, in _InternalImportModule
    parentModule, namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 423, in _LoadModule
    namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 494, in _LoadPackage
    self._LoadModule(name, fp, path, info, deferredImports, parent)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 473, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 589, in _ScanCode
    module, relativeImportIndex)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 309, in _ImportModule
    deferredImports, namespace = namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 410, in _InternalImportModule
    parentModule, namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 473, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 589, in _ScanCode
    module, relativeImportIndex)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 309, in _ImportModule
    deferredImports, namespace = namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 410, in _InternalImportModule
    parentModule, namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 473, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 589, in _ScanCode
    module, relativeImportIndex)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 309, in _ImportModule
    deferredImports, namespace = namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 410, in _InternalImportModule
    parentModule, namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 473, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 589, in _ScanCode
    module, relativeImportIndex)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 309, in _ImportModule
    deferredImports, namespace = namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 410, in _InternalImportModule
    parentModule, namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 473, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 613, in _ScanCode
    topLevel = False)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 613, in _ScanCode
    topLevel = False)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 589, in _ScanCode
    module, relativeImportIndex)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 309, in _ImportModule
    deferredImports, namespace = namespace)
  File "/home/sai/.local/lib/python3.5/site-packages/cx_Freeze/finder.py", line 398, in _InternalImportModule
    except (AttributeError, ModuleNotFoundError, ValueError):
NameError: name 'ModuleNotFoundError' is not defined

```
**It throws NameError module not found.**


**To conclude:**
- the setup.py worked well with cx freeze 6.1, but looks like it does not support pygame
- cx freeze 6.2 is throwing error module not found
- cz freze 6.3, strangely instead of building the test.py, it is executing test.py

**Comment by marcelotduarte** [comment]
Unfortunately, you are using very old systems and versions, you would not be able to test or fix any bugs.
However, rereading the topic, I see:
```
/home/sai/.local/lib/python3.5/site-packages/pygame/__init__.py", line 81, in <module>
    from pygame.base import * # pylint: disable=wildcard-import; lgtm[py/polluting-import]
ImportError: libSDL2-2-7159ff74.0.so.0.16.0
```
This file is probably distributed with pygame, so see if it is in some pygame subdirectory at /home/sai/.local/lib/python3.5/site-packages/pygame

If you find it use on of the keys [bin_includes](https://cx-freeze.readthedocs.io/en/latest/setup_script.html#cmdoption-arg-bin_includes) or [include_files](https://cx-freeze.readthedocs.io/en/latest/setup_script.html#cmdoption-arg-include_files).

**Comment by mai1x9** [comment]
@marcelotduarte Thank you very much, the trick has worked

I have modified options, in setup.py and included the below code
```
options = {
        'build_exe': {
            'packages': packages,
            'include_files' : [("/usr/local/lib/python3.5/dist-packages/pygame.libs", "lib/pygame.libs")],
        },
```
**I have moved the pygame.libs folder which has all so files, to lib folder and it has worked.**

Thank you very much.

But I wonder pygame folder has all so files which are present in pygame.libs folder, but still it has thrown error. When I moved pygame.libs folder to /lib it has worked.
**In fact pygame folder has libSDL .so file in it, still it couldn't figure out the .so file in its own directory, rather looking up in pygame.libs folder.**

**Any idea for this weird behaviour?**
